### PR TITLE
feat(catalog): Add Open Liberty support

### LINKED ISSUE #1008: Add Open Liberty as a supported backend
### Description

Open Liberty should be a supported backend in the launcher application. 

### What should happen?

Open Liberty should be available as a backend in the "Create a New Application" section. It should support health check, rest, and database-crud capabilities. 

### What happens instead?

Open Liberty is currently only supported in the "Deploy an Example application" section. 

### Do you have any idea why?

Not added yet.

### PR DESCRIPTION
**Pull request by brenthdaniel**
Adds Open Liberty as a supported backend.

Closes #1008

### Why

Adds capability

### Description

Adds Open Liberty as a supported backend runtime with support for health check, rest, and database-crud capabilities

Closes #1008

### Checklist

- [x ] I followed the contribution [guidelines](https://raw.githubusercontent.com/fabric8-launcher/launcher-application/master/CONTRIBUTING.md).
- [x ] I created an [issue](https://github.com/fabric8-launcher/launcher-application/issues/new) and used it in the commit message.
- [x ] I have built the project locally prior to submission with `mvn clean install`.
- [x ] I kept a clean commit log (no unnecessary commits, no merge commits).
- [x ] I am happy with my contribution.

### CODE REVIEW

**Comment by brenthdaniel** [comment]
@quintesse Will you be able to look at this review soon?

**Comment by quintesse** [comment]
@brenthdaniel I'm running the integration tests right now. If they pass things could be merged.

**Comment by quintesse** [comment]
I'll not some more time for this. Building any OpenLiberty app with S2I takes over 30 minutes on my machine and when it finally finishes I get OOM errors. When I work around those I get failed IT tests, but I can't be sure they are not related to the OOM workarounds.

Those 30+ minute build times worry me though.

**Comment by brenthdaniel** [comment]
yes.. I think most of the issue was that it was pulling every artifact from three separate maven repositories. The other s2i images seem to have a settings.xml that prevents that from happening. I updated the OpenLiberty S2I image, so things should be much faster now.

**Comment by quintesse** [comment]
Ok great, let me try again

**Comment by quintesse** [comment]
So I figured out that using CRC to run the tests is basically impossible. It's too much for my poor laptop.
But I was able to run them on an internal cluster. 4m build times is a lot more acceptable :-)

But the database tests when using PostgreSQL fail for me. Were you able to tests those successfully yourself @brenthdaniel ?

**Comment by brenthdaniel** [comment]
Yes... Not sure if this is what you're running into, but I think there may be a bit of a timing issue because Liberty can make the port available before the application has started. waitForAppResponse is successful as soon as it gets a 404, but the app may not be ready to run tests yet.

**Comment by quintesse** [comment]
In case you're wondering @brenthdaniel , I'm still working on this. Things look good on your side, no worries, but I just can't get a decent test setup running on my system. Thanks for your patience so far.

**Comment by quintesse** [comment]
> a 200 from an app specific URL rather than any response code between 200 and 500 from the app.

Thing is that it's not a given that a 200 is actually the correct response code. Many tests actually test for failure states.

So we might have to see if we can always safely do what you first suggested:

> the test waited on a health check URL

**Comment by brenthdaniel** [comment]
Thanks @quintesse . Let me know if there's anything I can help with.

**Comment by quintesse** [comment]
So I was finally able to make all the tests pass successfully. Thanks for your contribution @brenthdaniel ! (and your patience)
